app/controlplane/internal/usercontext/apitoken_middleware.go

```diff
@@ -101,6 +102,72 @@ func WithCurrentAPITokenAndOrgMiddleware(apiTokenUC *biz.APITokenUseCase, orgUC
 	}
 }
 
+// WithAttestationContextFromAPIToken injects the API-Token, organization + robot account to the context
+func WithAttestationContextFromAPIToken(apiTokenUC *biz.APITokenUseCase, logger *log.Helper) middleware.Middleware {
+	return func(handler middleware.Handler) middleware.Handler {
+		return func(ctx context.Context, req interface{}) (interface{}, error) {
+			authInfo, ok := attjwtmiddleware.FromJWTAuthContext(ctx)
+			// If not found means that there is no currentUser set in the context
+			if !ok {
+				logger.Warn("couldn't extract org/user, JWT parser middleware not running before this one?")
+				return nil, errors.New("can't extract JWT info from the context")
+			}
+
+			// If the token is not an API token, we don't need to do anything
+			if authInfo.ProviderKey != attjwtmiddleware.APITokenProviderKey {
+				return handler(ctx, req)
+			}
+
+			genericClaims, ok := authInfo.Claims.(jwt.MapClaims)
+			if !ok {
+				return nil, errors.New("error mapping the claims")
+			}
+
+			// We've received an API-token
+			if genericClaims.VerifyAudience(apitoken.Audience, true) {
```

Comment:
shouldn't this fail otherwise?

Reply:
Actually this check here can be deleted. The audience is already check previously on the `JWTMulti` middleware and we should be able to reach this point if and only if, the audience is valid.

I'll return an error.

Reply:
It's ok to keep it in two places. I lean towards defensive programming for these core pieces.